archive/100-theorems-list/81_sum_of_prime_reciprocals_diverges.lean

```diff
@@ -0,0 +1,371 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import data.nat.prime
+import data.nat.sqrt
+import data.real.basic
+import data.finset
+import order.filter
+import topology.basic
+import algebra.big_operators.ring
+import data.real.nnreal
+import topology.instances.ennreal
+import tactic.ring_exp
+import algebra.squarefree
+import data.pnat.factors
+
+/-!
+# Divergence of the Prime Reciprocal Series
+This file proves Theorem 81 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+The theorem states that the sum of the reciprocals of all prime numbers diverges.
+The formalization follows the Erdős's proof by upper and lower estimates.
+
+## References
+https://en.wikipedia.org/wiki/Divergence_of_the_sum_of_the_reciprocals_of_the_primes
+-/
+
+open_locale big_operators
+open_locale classical
+open nat pnat filter finset
+
+lemma nat.succ_pred {n : ℕ} (h : n > 0) : n.pred.succ = n :=
+begin
+  cases n,
+  { exfalso, exact nat.lt_asymm h h },
+  rw nat.pred_succ n,
+end
+
+lemma sq_mul_sq_free {n : ℕ} (hn : n > 0) :
```

Comment:
I think `hn` should be `0 < n`, and similarly for the rest of the file (unless this is not as big of a deal in `archive`)